tests/python/unittest/test_operator.py

```diff
@@ -6606,6 +6606,31 @@ def test_slice_forward_backward(a, index):
     for index in index_list:
         test_slice_forward_backward(arr, index)
 
+    def test_begin_equals_end():
+        shape = (2, 4)
+        begin = (None, 2)
+        end = (None, 2)
+        step = (1, -1)
+        in_arr = mx.nd.arange(np.prod(shape)).reshape(shape=shape)
+        out_arr = mx.nd.slice(in_arr, begin=begin, end=end, step=step)
+
+    assertRaises(MXNetError, test_begin_equals_end)
+
+    # check for end=-1 and negative step
```

Comment:
Can you also test end < -1 as in the branch on line https://github.com/apache/incubator-mxnet/pull/14403/files#diff-d5c9c7d539907caef01a08a74e16592bR689